lib/templating.sh

```diff
@@ -20,25 +20,46 @@ function _perform_templating() {
   local i
   local -a _keys
   # Shorter path comes earlier than longer.
-  mapfile -t _keys < <(_paths_of_string_type "${_content}")
-  for i in "${_keys[@]}"; do
-    local _node_value _templated_node_value _ret_file
-    _node_value="$(value_at "${i}" "${_ret}")"
-    _ret_file=$(mktemp_with_content "${_ret}")
-    _templated_node_value="$(_render_text_node "${_node_value}" "${i}" "${_ret_file}")"
-    _ret="$(jq -r -c -n "input|${i}=input" <(echo "${_ret}") <(echo "${_templated_node_value}"))"
+  while true; do
+    mapfile -t _keys < <(_paths_of_string_nodes_perform_templating "${_ret}")
+    if is_empty_array "${_keys[@]}"; then
+      break
+    fi
+    debug "begin loop"
```

Comment:
We should have a check for nested templating level.
Otherwise, the templating may happen infinitely.
The level should be a confgurable parameter, whose default is a decent constant (probably 6 or 7?)

Reply:
Addressed.
I made the value 5.
Since it turned out to be sufficient.